### PR TITLE
Add network test

### PR DESCRIPTION
**Pull request by asdil12**
Ticket: https://progress.opensuse.org/issues/43346
Verification:

TW Gnome: http://artemis.suse.de/tests/813
TW Text: http://artemis.suse.de/tests/814
SLE15SP1 Text: http://artemis.suse.de/tests/818

### CODE REVIEW

File: tests/console/network.pm [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# Maintainer: Dominik Heidler <dheidler@suse.de>
+
+use base 'consoletest';
+use strict;
+use testapi;
+use utils;
+use version_utils;
+
+sub run {
+    select_console 'root-console';
```
Comment (asmorodskyi) [inline tests/console/network.pm]: why not to use serial terminal here ?

Comment (asdil12) [reply tests/console/network.pm]: Why should I?

Comment (asdil12) [reply tests/console/network.pm]: Well - actually I am using the serial line for script return status.
Like any other `console` test is doing it.

File: tests/console/network.pm [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    if (is_opensuse) {
+        # check for systemd-networkd
+        systemctl 'is-enabled systemd-networkd', expect_false => 1;
+        systemctl 'is-active systemd-networkd',  expect_false => 1;
+        assert_script_run 'networkctl status';
+    }
+
+    my $network_daemon = script_output 'readlink /etc/systemd/system/network.service | sed \'s#.*/\(.*\)\.service#\1#\'';
+
+    if (check_var('DESKTOP', 'textmode') && $network_daemon ne 'wicked') {
```
Comment (rwx788) [inline tests/console/network.pm]: What's about rewriting it as follows not to do same checks twice:
```perl
    if (check_var('DESKTOP', 'textmode')) {
        die "Expected wicked for DESKTOP=textmode but got '$network_daemon'" if $network_daemon ne 'wicked';       
    }
    else {
        die "Expected NetworkManager for DESKTOP!=textmode but got '$network_daemon'" if $network_daemon ne 'NetworkManager';
    }
```
Also, I thought for SLES we always get wicked, so maybe we need to inject more complex condition for expectations. NM is default for SLED though.

Comment (asdil12) [reply tests/console/network.pm]: Updated the patch as suggested.

File: tests/console/network.pm [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        die "Expected wicked for DESKTOP=textmode but got '$network_daemon'";
+    }
+    elsif (!check_var('DESKTOP', 'textmode') && $network_daemon ne 'NetworkManager') {
+        die "Expected NetworkManager for DESKTOP!=textmode but got '$network_daemon'";
+    }
+
+    record_info "$network_daemon";
```
[the remark this answers is not in the document]
Comment (asdil12) [reply tests/console/network.pm]: done.

File: tests/console/network.pm (hunk 2 above)
Comment (asdil12) [reply tests/console/network.pm]: Fixed the code to expect nm for SLED and wicked for SLES.

File: tests/console/network.pm [hunk 4]
```diff
@@ -0,0 +1,58 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Check for network daemon
```
[the remark this answers is not in the document]
Comment (asdil12) [reply tests/console/network.pm]: done

File: tests/console/network.pm (hunk 2 above)
Comment (asdil12) [reply tests/console/network.pm]: I'm just not sure where to schedule the test for SLED as SLED doesn't have `extra_tests`.
And how should we handle this with SLE15 where desktop will just be an addon?

@okurz WDYT?

File: tests/console/check_default_network_manager.pm [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+use testapi;
+use utils;
+use version_utils;
+
+sub run {
+    select_console 'root-console';
+
+    assert_script_run 'ip a';
+
+    if (is_opensuse) {
```
Comment (okurz) [inline tests/console/check_default_network_manager.pm]: could you explain why?

Comment (asdil12) [reply tests/console/check_default_network_manager.pm]: systemd on SLE is compiled without networkd support.

File: tests/console/check_default_network_manager.pm [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        systemctl 'is-active wicked';
+        systemctl 'status wicked';
+        assert_script_run 'wicked show all';
+    }
+    elsif ($network_daemon eq "NetworkManager") {
+        # check if NetworkManager is installed, enabled and running
+        assert_script_run 'rpm -q NetworkManager';
+        systemctl 'is-enabled NetworkManager';
+        systemctl 'is-active NetworkManager';
+        systemctl 'status NetworkManager';
```
[the remark this answers is not in the document]
Comment (asdil12) [reply tests/console/check_default_network_manager.pm]: done

**Pull request merged**